### ISSUE TITLE
brace-style rule with stroustrup does not report on cuddled else if

### ISSUE THREAD

**Issue by LewisJEllis** [eslint/eslint#1583]
With `brace-style: [2, "stroustrup"]`, the following snippet:

```
if (foo) {
  baz();
} else if (bar) {
  baz();
}
else {
  qux();
}
```

does not report an error for the `} else if {`, but I believe it should, in the same way that:

```
if (foo) {
  bar();
} else {
  baz();
}
```

reports an error for the `} else {`.

I believe Stroustrup's rules for not "cuddling the else" should also apply to else if, as shown [here](http://www.stroustrup.com/4th_printing3.html) (search "else if").
